from PyQt5 import uic
from PyQt5.QtWidgets import QApplication, QInputDialog

import json

Form, Window = uic.loadUiType("untitled.ui")

app = QApplication([])
window = Window()
form = Form()
form.setupUi(window)

'''notes = {
    "–î–æ–±—Ä–æ –ø–æ–∂–∞–ª–æ–≤–∞—Ç—å!" : {
        "—Ç–µ–∫—Å—Ç" : "–≠—Ç–æ —Å–∞–º–æ–µ –ª—É—á—à–µ–µ –ø—Ä–∏–ª–æ–∂–µ–Ω–∏–µ –¥–ª—è –∑–∞–º–µ—Ç–æ–∫ –≤ –º–∏—Ä–µ!",
        "—Ç–µ–≥–∏" : ["–¥–æ–±—Ä–æ", "–∏–Ω—Å—Ç—Ä—É–∫—Ü–∏—è"]
    }
}
with open("notes_data.json", "w") as file:
    json.dump(notes, file)'''

def add_note():
    note_name, ok = QInputDialog.getText(window, "–î–æ–±–∞–≤–∏—Ç—å –∑–∞–º–µ—Ç–∫—É", "–ù–∞–∑–≤–∞–Ω–∏–µ –∑–∞–º–µ—Ç–∫–∏: ")
    if ok and note_name != "":
        notes[note_name] = {"—Ç–µ–∫—Å—Ç" : "", "—Ç–µ–≥–∏" : []}
        form.SpisokZametok.addItem(note_name)
        form.SpisokZametok_2.addItems(notes[note_name]["—Ç–µ–≥–∏"])
        print(notes)

def show_note():
    key = form.SpisokZametok.selectedItems()[0].text()
    print(key)
    form.textEdit.setText(notes[key]["—Ç–µ–∫—Å—Ç"])
    form.SpisokZametok_2.clear()
    form.SpisokZametok_2.addItems(notes[key]["—Ç–µ–≥–∏"])

def save_note():
    if form.SpisokZametok.selectedItems():
        key = form.SpisokZametok.selectedItems()[0].text()
        notes[key]["—Ç–µ–∫—Å—Ç"] = form.textEdit.toPlainText()
        with open('notes_data.json', 'w') as file:
            json.dump(notes, file, sort_keys=True, ensure_ascii=False)
        print(notes)
    else:
        print('–ó–∞–º–µ—Ç–∫–∞ –¥–ª—è —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è –Ω–µ –≤—ã–±—Ä–∞–Ω–∞!')

def del_note():
    if form.SpisokZametok.selectedItems():
        key = form.SpisokZametok.selectedItems()[0].text()
        del notes[key]
        form.SpisokZametok.clear()
        form.SpisokZametok_2.clear()
        form.textEdit.clear()
        form.SpisokZametok.addItems(notes)
        with open('notes_data.json', 'w') as file:
            json.dump(notes, file, sort_keys=True, ensure_ascii=False)
        print(notes)
    else:
        print('–ó–∞–º–µ—Ç–∫–∞ –¥–ª—è —É–¥–∞–ª–µ–Ω–∏—è –Ω–µ –≤—ã–±—Ä–∞–Ω–∞!')

def add_tag():
    if form.SpisokZametok.selectedItems():
        key = form.SpisokZametok.selectedItems()[0].text()
        tag = form.lineEdit.text()
        if not tag in notes[key]["—Ç–µ–≥–∏"]:
            notes[key]["—Ç–µ–≥–∏"].append(tag)
            form.SpisokZametok_2.addItem(tag)
            form.lineEdit.clear()
        with open("notes_data.json", "w") as file:
            json.dump(notes, file, sort_keys=True, ensure_ascii=False)
        print(notes)
    else:
        print("–ó–∞–º–µ—Ç–∫–∞ –¥–ª—è –¥–æ–±–∞–≤–ª–µ–Ω–∏—è —Ç–µ–≥–∞ –Ω–µ –≤—ã–±—Ä–∞–Ω–∞!")

def del_tag():
    if form.SpisokZametok_2.selectedItems():
        key = form.SpisokZametok.selectedItems()[0].text()
        tag = form.SpisokZametok_2.selectedItems()[0].text()
        notes[key]["—Ç–µ–≥–∏"].remove(tag)
        form.SpisokZametok_2.clear()
        form.SpisokZametok_2.addItems(notes[key]["—Ç–µ–≥–∏"])
        with open("notes_data.json", "w") as file:
            json.dump(notes, file, sort_keys=True, ensure_ascii=False)
        print(notes)
    else:
        print("–¢–µ–≥ –¥–ª—è —É–¥–∞–ª–µ–Ω–∏—è –Ω–µ –≤—ã–±—Ä–∞–Ω!")

def search_tag():
    print(form.SearchZametka.text())
    tag = form.lineEdit.text()
    if form.SearchZametka.text() == "–ò—Å–∫–∞—Ç—å –∑–∞–º–µ—Ç–∫–∏ –ø–æ —Ç–µ–≥—É" and tag:
        print(tag)
        notes_filtered = {}
        for note in notes:
            if tag in notes [note]["—Ç–µ–≥–∏"]:
                notes_filtered[note]=notes[note]
        form.SearchZametka.setText("–°–±—Ä–æ—Å–∏—Ç—å –ø–æ–∏—Å–∫")
        form.SpisokZametok.clear()
        form.SpisokZametok_2.clear()
        form.SpisokZametok.addItems(notes_filtered)
        print(form.SearchZametka.text())
    elif form.SearchZametka.text() == "–°–±—Ä–æ—Å–∏—Ç—å –ø–æ–∏—Å–∫":
        form.lineEdit.clear()
        form.SpisokZametok.clear()
        form.SpisokZametok_2.clear()
        form.SpisokZametok.addItems(notes)
        form.SearchZametka.setText("üîé–ò—Å–∫–∞—Ç—å –∑–∞–º–µ—Ç–∫–∏ –ø–æ —Ç–µ–≥—Éüîé")
        print(form.SearchZametka.text())
    else:
        pass

form.SpisokZametok.itemClicked.connect(show_note)
form.SaveZametka.clicked.connect(save_note)
form.DelZametka.clicked.connect(del_note)
form.AddZametka.clicked.connect(add_note)
form.AddZametka_2.clicked.connect(add_tag)
form.DelZametka_2.clicked.connect(del_tag)
form.SearchZametka.clicked.connect(search_tag)
with open("notes_data.json", "r") as file:
    notes = json.load(file)
form.SpisokZametok.addItems(notes)
window.show()
app.exec()
